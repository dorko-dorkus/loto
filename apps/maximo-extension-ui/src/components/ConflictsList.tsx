import React, { useState } from 'react';
import { fetchRecommendedSet } from '../mocks/bundling';

interface Candidate {
  id: string;
  label: string;
  explanation: string;
}

interface ConflictsListProps {
  candidates: Candidate[];
}

export default function ConflictsList({ candidates }: ConflictsListProps) {
  const [selected, setSelected] = useState<string[]>([]);

  const toggle = (id: string) => {
    setSelected((prev) =>
      prev.includes(id) ? prev.filter((x) => x !== id) : [...prev, id]
    );
  };

  const handleRecommend = async () => {
    const rec = await fetchRecommendedSet();
    setSelected(rec);
  };

  return (
    <div>
      <ul>
        {candidates.map(({ id, label, explanation }) => (
          <li key={id} className="mb-2">
            <label className="flex items-center gap-2">
              <input
                type="checkbox"
                checked={selected.includes(id)}
                onChange={() => toggle(id)}
                aria-label={label}
              />
              <span>{label}</span>
            </label>
            <p className="ml-6 text-sm text-gray-600">{explanation}</p>
          </li>
        ))}
      </ul>
      <button type="button" aria-label="Select recommended set" onClick={handleRecommend}>
        Select recommended set
      </button>
    </div>
  );
}
