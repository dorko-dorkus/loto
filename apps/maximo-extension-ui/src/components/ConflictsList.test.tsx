import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { test, expect } from 'vitest';
import ConflictsList from './ConflictsList';

test('click sets recommended items', async () => {
  const candidates = [
    { id: '1', label: 'Candidate 1', explanation: 'First option' },
    { id: '2', label: 'Candidate 2', explanation: 'Second option' },
    { id: '3', label: 'Candidate 3', explanation: 'Third option' }
  ];
  const { container } = render(<ConflictsList candidates={candidates} />);
  fireEvent.click(screen.getByText('Select recommended set'));
  await waitFor(() => {
    expect(screen.getByLabelText('Candidate 1')).toBeChecked();
    expect(screen.getByLabelText('Candidate 3')).toBeChecked();
  });
  expect(container.firstChild).toMatchSnapshot();
});
