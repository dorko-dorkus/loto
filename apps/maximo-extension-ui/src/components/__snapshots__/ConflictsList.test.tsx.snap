// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`click sets recommended items 1`] = `
<div>
  <ul>
    <li
      class="mb-2"
    >
      <label
        class="flex items-center gap-2"
      >
        <input
          aria-label="Candidate 1"
          type="checkbox"
        />
        <span>
          Candidate 1
        </span>
      </label>
      <p
        class="ml-6 text-sm text-gray-600"
      >
        First option
      </p>
    </li>
    <li
      class="mb-2"
    >
      <label
        class="flex items-center gap-2"
      >
        <input
          aria-label="Candidate 2"
          type="checkbox"
        />
        <span>
          Candidate 2
        </span>
      </label>
      <p
        class="ml-6 text-sm text-gray-600"
      >
        Second option
      </p>
    </li>
    <li
      class="mb-2"
    >
      <label
        class="flex items-center gap-2"
      >
        <input
          aria-label="Candidate 3"
          type="checkbox"
        />
        <span>
          Candidate 3
        </span>
      </label>
      <p
        class="ml-6 text-sm text-gray-600"
      >
        Third option
      </p>
    </li>
  </ul>
  <button
    aria-label="Select recommended set"
    type="button"
  >
    Select recommended set
  </button>
</div>
`;
